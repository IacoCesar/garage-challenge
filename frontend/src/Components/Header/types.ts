export interface IHeader {
  testID?: string
  isHomeScreen?: boolean
  title?: string
  callHandleBack?: (navigation: any) => void
  callHandleChangeVisualMode?: (themeSelector: string) => void
}

export interface IProps extends IHeader {
  themeSelector: string
}
