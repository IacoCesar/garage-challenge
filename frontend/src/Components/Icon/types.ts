export interface IProps {
  testID?: string
  size: number
  name: string
  color: string
  type?: string
}
