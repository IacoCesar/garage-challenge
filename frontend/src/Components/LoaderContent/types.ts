export interface IProps {
  testID?: string
  size: number
  color: string
}
