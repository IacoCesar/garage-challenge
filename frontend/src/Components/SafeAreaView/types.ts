export interface IProps {
  testID?: string
  children: JSX.Element
}

export interface IThemeProps {
  themeSelector: string
}
