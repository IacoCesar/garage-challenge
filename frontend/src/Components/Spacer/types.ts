export interface IProps {
  testID?: string
  min: number
  max: number
}
