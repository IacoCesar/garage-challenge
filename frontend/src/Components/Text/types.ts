export interface ITextProps {
  color?: string
  fontSize?: number
  isBold?: boolean
}

export interface IProps extends ITextProps {
  label: string
  testID?: string
}
