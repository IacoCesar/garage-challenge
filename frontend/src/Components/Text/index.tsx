import { assignTestId } from '@/Helpers/TestId'
import { RootState } from '@/Store'
import React from 'react'
import { useSelector } from 'react-redux'
import { useTheme } from 'styled-components/native'
import S from './styles'
import { IProps } from './types'

const Component: React.FC<IProps> = ({
  testID = 'Text',
  label,
  fontSize,
  color,
  isBold
}) => {
  const themeSelector = useSelector((state: RootState) => state.themeData.theme)
  const theme = useTheme()

  return (
    <S.Content {...assignTestId('View', `${testID}`)}>
      <S.Text
        {...assignTestId('Text', `${testID}_Content_Text`)}
        fontSize={fontSize}
        color={color || theme.standard[themeSelector].textColor}
        isBold={isBold}
      >
        {label}
      </S.Text>
    </S.Content>
  )
}

export default Component
