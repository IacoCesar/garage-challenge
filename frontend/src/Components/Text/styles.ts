import styled from 'styled-components/native'
import { ITextProps } from './types'

const Content = styled.View`
  display: flex;
`

const Text = styled.Text<ITextProps>`
  font-size: ${({ fontSize }) => fontSize || `12px`};
  color: ${({ theme, color }) => color || theme.standard.darkMode.textColor};
  ${({ isBold }) => isBold && `font-weight: bold;`};
`

export default {
  Text,
  Content
}
