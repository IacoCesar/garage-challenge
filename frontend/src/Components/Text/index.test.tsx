import React from 'react'
import { IProps } from './types'
import { mockTestId } from '@/Helpers/TestId'
import Text from '.'
import { render } from '@/Helpers/TestHelpers'

describe('Text', () => {
  let props: IProps = {
    testID: 'Text',
    label: 'Test'
  }

  const loaderElement = mockTestId('View', `${props.testID!}`).testID

  it(`should render ${loaderElement}`, async () => {
    const wrapper = render(<Text {...props} />)

    const el_container = wrapper.getByTestId(loaderElement)
    expect(el_container).toBeTruthy()
  })

  it('should render snapshot', async () => {
    const wrapper = render(<Text {...props} />)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
