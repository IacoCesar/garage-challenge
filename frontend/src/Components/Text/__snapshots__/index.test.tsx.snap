// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render snapshot 1`] = `
<View
  accessible={false}
  style={
    Array [
      Object {
        "display": "flex",
      },
    ]
  }
  testID="View_Text"
>
  <Text
    accessible={false}
    color="#000000"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontSize": 12,
        },
      ]
    }
    testID="Text_Text_Content_Text"
  >
    Test
  </Text>
</View>
`;
